iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/parser/SpringConfigParser.java

```diff
@@ -19,18 +19,38 @@
  */
 package org.sonar.iac.springconfig.parser;
 
-import java.util.List;
 import javax.annotation.Nullable;
 import org.sonar.iac.common.api.tree.Tree;
+import org.sonar.iac.common.extension.ParseException;
 import org.sonar.iac.common.extension.TreeParser;
 import org.sonar.iac.common.extension.visitors.InputFileContext;
+import org.sonar.iac.common.yaml.YamlLanguage;
+import org.sonar.iac.springconfig.parser.properties.SpringConfigPropertiesParser;
+import org.sonar.iac.springconfig.parser.yaml.SpringConfigYamlParser;
 import org.sonar.iac.springconfig.tree.api.SpringConfig;
-import org.sonar.iac.springconfig.tree.impl.FileImpl;
 
 public class SpringConfigParser implements TreeParser<Tree> {
+
+  private static final SpringConfigYamlParser YAML_PARSER = new SpringConfigYamlParser();
+  private static final SpringConfigPropertiesParser PROPERTIES_PARSER = new SpringConfigPropertiesParser();
+
   @Override
   public SpringConfig parse(String source, @Nullable InputFileContext inputFileContext) {
-    // TODO: SONARIAC-1446 Implement "SpringConfigParser"
-    return new FileImpl(List.of());
+    if (inputFileContext != null) {
+      if (isYamlFile(inputFileContext)) {
+        return YAML_PARSER.parse(source, inputFileContext);
+      } else if (isPropertiesFile(inputFileContext)) {
+        return PROPERTIES_PARSER.parse(source, inputFileContext);
+      }
+    }
+    throw ParseException.createParseException("Unsupported file extension", inputFileContext, null);
+  }
+
+  private static boolean isYamlFile(InputFileContext inputFileContext) {
+    return YamlLanguage.KEY.equals(inputFileContext.inputFile.language());
+  }
+
+  private static boolean isPropertiesFile(InputFileContext inputFileContext) {
```

Comment:
There is already a method like this in `SpringConfigSensor`. Maybe it can be kept in a single place?

Reply:
Ok, I will move the logic to the Sensor